ste/pacer-tokenBucketPacer.go

```diff
@@ -66,23 +85,44 @@ func newTokenBucketPacer(ctx context.Context, bytesPerSecond int64, expectedByte
 	return p
 }
 
-// RequestRightToSend function is called by goroutines to request right to send a certain amount of bytes.
+// RequestTrafficAllocation function is called by goroutines to request right to send a certain amount of bytes.
 // It controls their rate by blocking until they are allowed to proceed
-func (p *tokenBucketPacer) RequestRightToSend(ctx context.Context, bytesToSend int64) error {
-	for atomic.AddInt64(&p.atomicTokenBucket, -bytesToSend) < 0 {
+func (p *tokenBucketPacer) RequestTrafficAllocation(ctx context.Context, byteCount int64) error {
```

Comment:
I'll admit, this entire setup is really clever, basically using max throughput as a currency of sorts. I've got to wonder though, would it be theoretically better to pace disk usage alongside pacing network usage? I could be dead wrong on how we handle disk usage. My thinking is that disk-usage wise, we could limit reads to a buffer slightly larger than the actual network output (so that we have a buffer zone for disk latency), and thus be a good neighbor on the local machine.

Reply:
Do you mean, "We should limit disk usage to avoid conflicting with other things that also need _disk_" or "We should limit disk usage as a preferrable implementation of controlling _network_ throughput" or "We should do both, because doing just one won't be enough"?   

I must admit I never thought of using a disk throughput control as the sole method of controlling network throughput. I think it has a theoretical disadvantage from the network perpective, which is that, as you say, there would have to be a buffer. In the cases where that buffer is non-empty, our control over network throughput would be much looser, and so network throughput could be more spikey.  My feeling is that smooth network throughput (at the expense of spikey disk) is better than smooth disk throughput (at the expense of spikey network).  For the reason outlined below, I don't think its necessary to do both.

I guess the other question is whether a user concerned about disk throughput, could reduce AzCopy's disk throughput by using the _network_ cap as currently implement.  And the answer is, yes they could.  If you set a low network cap, our disk buffers will soon reach their limit, and your disk throughput will slow to whatever you set as the network cap.

Reply:
I meant the former, rather than the latter.

Reply:
I think we can let folks do that just by using the existing flag as implemented. (As per final paragraph of my comment above). I think we don't want to have _two_ flags - that would just be confusing because only the lower of the two would actually have any meaningful effect.  On that basis I'm marking this thread as resolved, but if that hasn't adequately answered your question please unresolved it and elaborate in a comment.